api/apps/geoprocessing/src/modules/scenario-planning-units-inclusion/scenario-planning-units-inclusion-processor.ts

```diff
@@ -84,45 +89,89 @@ export class ScenarioPlanningUnitsInclusionProcessor
       );
     }
 
+    const puIdsToMakeAvailableFromGeo: string[] = [];
+    if (makeAvailableGeo) {
+      const targetGeometries = flatMap(
+        makeAvailableGeo,
+        (collection) => collection.features,
+      ).map((feature) => feature.geometry);
+      puIdsToMakeAvailableFromGeo.push(
+        ...(
+          await this.getPlanningUnitsIntersectingGeometriesFor(
+            scenarioId,
+            targetGeometries,
+          )
+        ).map(({ spd_id: id }) => id),
+      );
+    }
+
     const puIdsToIncludeFromIds = job.data.include?.pu ?? [];
     const puIdsToExcludeFromIds = job.data.exclude?.pu ?? [];
-
-    // If there are overlaps between opposing claims byId and byGeoJSON, ignore the claims byId
-    const puIdsToIncludeFromIdsLessIdsToExcludeFromGeo = difference(
-      puIdsToIncludeFromIds,
-      puIdsToExcludeFromGeo,
-    );
-    const puIdsToExcludeFromIdsLessIdsToIncludeFromGeo = difference(
-      puIdsToExcludeFromIds,
-      puIdsToIncludeFromGeo,
-    );
+    const puIdsToMakeAvailableFromIds = job.data.makeAvailable?.pu ?? [];
 
     // Union of claims byId and byGeoJSON, for inclusions and for exclusions
     puIdsToInclude.push(
-      ...[
-        ...puIdsToIncludeFromIdsLessIdsToExcludeFromGeo,
-        ...puIdsToIncludeFromGeo,
-      ],
+      ...[...puIdsToIncludeFromIds, ...puIdsToIncludeFromGeo],
     );
     puIdsToExclude.push(
-      ...[
-        ...puIdsToExcludeFromIdsLessIdsToIncludeFromGeo,
-        ...puIdsToExcludeFromGeo,
-      ],
+      ...[...puIdsToExcludeFromIds, ...puIdsToExcludeFromGeo],
+    );
+
+    puIdsToMakeAvailable.push(
+      ...[...puIdsToMakeAvailableFromIds, ...puIdsToMakeAvailableFromGeo],
     );
     const uniquePuIdsToInclude = new Set(puIdsToInclude);
     const uniquePuIdsToExclude = new Set(puIdsToExclude);
+    const uniquePuIdsToMakeAvailable = new Set(puIdsToMakeAvailable);
 
     const doInclusionAndExclusionIntersect =
       intersection([...uniquePuIdsToInclude], [...uniquePuIdsToExclude])
         .length > 0;
 
+    const doMakeAvailableAndExclusionIntersect =
+      intersection([...uniquePuIdsToMakeAvailable], [...uniquePuIdsToExclude])
+        .length > 0;
+
+    const doMakeAvailableAndInclusionIntersect =
+      intersection([...uniquePuIdsToMakeAvailable], [...uniquePuIdsToInclude])
+        .length > 0;
+
```

Comment:
all we need to check is for any intersections/overlaps, and inform the user.

in practice, if the request comes from the frontend, this will be unlikely to happen because of the way the UX works, so only letting the user know that there is an overlap of some kind is ok, in my opinion, without having to tell them between which kinds of claims (besides, there could be multiple intersections between all the kinds, but we'd only let the user know about the first one we check, in the error handling below)